es-vue-base/src/lib-components/EsVideo.vue

```diff
@@ -0,0 +1,95 @@
+<template>
+    <div>
+        <b-embed
+            v-if="showVideo"
+            allowfullscreen
+            aspect="16by9"
+            :src="autoplayUrl"
+            :title="altText"
+            type="iframe" />
+        <es-button
+            v-else
+            class="EsVideo-button text-gray-900 border-0 p-0 position-relative text-gray w-100"
+            @click="showVideo = true">
+            <slot
+                v-if="hasImage"
+                name="image" />
+            <b-img
+                v-else-if="altText && coverImageUrl"
+                :alt="altText"
+                sizes="md:530px sm:275px"
+                class="EsVideo-image d-block rounded-lg w-100"
+                :src="coverImageUrl" />
+            <icon-video-play
+                class="EsVideo-icon position-absolute"
+                width="74px"
+                height="54px" />
+        </es-button>
+    </div>
+</template>
+
+<script lang="js">
+
+import { BImg, BEmbed } from 'bootstrap-vue';
+import IconVideoPlay from '../lib-icons/icon-video-play.vue';
+
+export default ({
+    name: 'EsVideo',
+    components: {
+        BImg,
+        BEmbed,
+        IconVideoPlay,
+    },
+    props: {
+        // eslint-disable-next-line vue/require-default-prop
+        altText: {
+            type: String,
+            required: false,
```

Comment:
I think for page load performance purposes, we're going to want a cover image in all cases, so we should probably make `altText` and `coverImageUrl` required.